### PR TITLE
XPU Support external stream

### PR DESCRIPTION
**Pull request by csy0225**
### PR types

New features
### PR changes

 APIs
### Description

XPU Support external stream.

### CODE REVIEW

**Review by jiweibo — APPROVED** [review]
`ResourceManager`部分后续可能需修改

https://github.com/PaddlePaddle/Paddle/pull/51412

**Review by zhupengyang — APPROVED** [review]
LGTM

File: paddle/phi/backends/xpu/xpu_context.cc [hunk 1]
```diff
@@ -246,6 +256,20 @@ const Place& XPUContext::GetPlace() const { return impl_->GetPlace(); }
...
+void XPUContext::SetXpuVersion(int version) {
+  impl_->xpu_version_ = static_cast<backends::xpu::XPUVersion>(version);
+}
+
+void XPUContext::SetRuntimeVersion(int version) {
+  impl_->runtime_version_ = version;
+}
+
+void XPUContext::SetDriverVersion(int version) {
+  impl_->driver_version_ = version;
```
Comment (hong19860320) [inline paddle/phi/backends/xpu/xpu_context.cc:270]: 后面是否需要补充获取 XPU driver or runtime version？

**Review by hong19860320 — COMMENTED** [review]
LGTM

File: paddle/phi/backends/xpu/xpu_context.cc (hunk 1 above)
Comment (csy0225) [reply paddle/phi/backends/xpu/xpu_context.cc:270]: 这个地方可以做兼容性判断，就看后续需不需要了

**Pull request merged**